crates/apollo-compiler/src/parser.rs

```diff
@@ -23,17 +23,6 @@ pub struct Parser {
     tokens_reached: usize,
 }
 
-/// Records for validation information about a file that was parsed
-#[derive(Clone)]
-pub struct SourceFile {
```

Comment:
(It’s only a mild preference, but I think grouping type definitions at the start of a module helps looking at them together instead of hunting for them in the middle of `impl` blocks that can be much longer. Data structures are essential.)

Reply:
omg i am the complete opposite haha!! i feel like i always need my structs with my impls so i can look up to see what i am working with 😅 😅 

i'll move this one back up since it seems like everything else uses your system